motr/idx_dix.c

```diff
@@ -549,17 +549,38 @@ static void dix_build(const struct m0_op_idx *oi,
 {
 	unsigned int   opcode = OP_IDX2CODE(oi);
 	struct m0_idx *idx = oi->oi_idx;
+	int            rc;
 
 	M0_SET0(out);
 	out->dd_fid = *OI_IFID(oi);
 	/* Pool version and layout type which are passed by consumers like S3 */
 	if (M0_IN(opcode, (M0_IC_GET, M0_IC_PUT, M0_IC_DEL, M0_IC_NEXT))) {
-		if (idx->in_attr.idx_layout_type == DIX_LTYPE_DESCR) {
-			M0_LOG(M0_DEBUG, "Opcode: %u, DIX pool version:"FID_F"",
+		if ((idx->in_attr.idx_layout_type == DIX_LTYPE_DESCR) 
+		    && (m0_fid_is_set(&idx->in_attr.idx_pver))
+		    && (m0_fid_is_valid(&idx->in_attr.idx_pver))) {
+			M0_LOG(M0_ALWAYS, "Opcode: %u, DIX pool version:"FID_F"",
 			       opcode, FID_P(&idx->in_attr.idx_pver));
-			out->dd_layout.dl_type = DIX_LTYPE_DESCR;
-			out->dd_layout.u.dl_desc.ld_pver = idx->in_attr.idx_pver;
+
+			rc = m0_dix_imask_init(&out->dd_layout.u.dl_desc.ld_imask, 
+			     &(struct m0_ext) { .e_start = 0, .e_end = IMASK_INF } , 1);
+			if (rc == 0) {	
+				out->dd_layout.dl_type = DIX_LTYPE_DESCR;
+				out->dd_layout.u.dl_desc.ld_pver = idx->in_attr.idx_pver;
+				out->dd_layout.u.dl_desc.ld_hash_fnc = HASH_FNC_CITY;
+			}
```

Comment:
I think we can use m0_dix_ldesc_init(), because it will call m0_dix_imask_init() and do the same.

Reply:
Done.